boa/src/syntax/lexer/spread.rs

```diff
@@ -0,0 +1,41 @@
+use super::{Cursor, Error, Tokenizer};
+use crate::syntax::ast::{Position, Punctuator, Span};
+use crate::syntax::lexer::Token;
+use std::io::Read;
+
+/// String literal lexing.
+///
+/// Note: expects for the initializer `'` or `"` to already be consumed from the cursor.
+#[derive(Debug, Clone, Copy)]
+pub(super) struct SpreadLiteral;
+
+impl SpreadLiteral {
+    /// Creates a new string literal lexer.
+    pub(super) fn new() -> Self {
+        Self {}
+    }
+}
+
+impl<R> Tokenizer<R> for SpreadLiteral {
+    fn lex(&mut self, cursor: &mut Cursor<R>, start_pos: Position) -> Result<Token, Error>
+    where
+        R: Read,
+    {
+        // . or ...
+        match cursor.next_is('.') {
```

Comment:
I think this can be done easier by using `if cursor.next_is('.')? {`.

Reply:
Yes definitely